src/libcore/cell.rs

```diff
@@ -798,6 +798,15 @@ impl<'b, T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<RefMut<'b, U>> for RefM
 /// The `UnsafeCell<T>` type is the only legal way to obtain aliasable data that is considered
 /// mutable. In general, transmuting an `&T` type into an `&mut T` is considered undefined behavior.
 ///
+/// The compiler makes optimizations based on the knowledge that `&T` is not mutably aliased, and
+/// that `&mut T` is unique. When building abstractions like `Cell`, `RefCell`, `Mutex`, etc,
+/// you need to turn these optimizations off. `UnsafeCell` is the way to do this,
```

Comment:
If we change this into "`UnsafeCell` is the only legal way to do this:" we can remove one more sentence above your addition. I like this version better.